projects/core/src/state/utils/get-state-slice.ts

```diff
@@ -31,17 +39,74 @@ export function createShellObject<T, E>(key: string, value: T): E {
   return newObject as E;
 }
 
-export function getStateSlice<T, E>(keys: string[], state: T): E {
+export function handleExclusions<T>(
+  key: string,
+  excludeKeys: string[],
+  value: T
+): T {
+  const { exclude, exclusionKey } = shouldExclude(key, excludeKeys);
+  if (!exclude) {
+    return value;
+  }
+
+  const exclusionChunksSplit = exclusionKey
+    .substring(key.length + 1, exclusionKey.length)
+    .split(OBJECT_SEPARATOR);
+
+  const finalValue = deepMerge({}, value);
+  let nestedTemp = finalValue;
+  for (let i = 0; i < exclusionChunksSplit.length; i++) {
+    const currentChunk = exclusionChunksSplit[i];
+
+    // last iteration
+    if (i === exclusionChunksSplit.length - 1) {
+      delete nestedTemp[currentChunk];
+    } else {
+      nestedTemp = nestedTemp[currentChunk];
+      if (!nestedTemp || Object.keys(nestedTemp).length === 0) {
+        return value;
+      }
+    }
+  }
+
+  return finalValue;
+}
+
+export function getStateSlice<T, E>(
+  keys: string[],
+  excludeKeys: string[],
+  state: T
+): E {
   if (keys && keys.length === 0) {
     return {} as E;
   }
 
   let stateSlices = {};
   for (const currentKey of keys) {
     const stateValue = getStateSliceValue(currentKey, state);
-    const shell = createShellObject(currentKey, stateValue);
+    const shell = createShellObject(currentKey, excludeKeys, stateValue);
     stateSlices = deepMerge(stateSlices, shell);
   }
 
   return stateSlices as E;
 }
+
+export function shouldExclude(
+  key: string,
+  excludeKeys: string[]
+): {
+  exclude: boolean;
+  exclusionKey: string;
+} {
+  if (!key || !excludeKeys) {
+    return { exclude: false, exclusionKey: '' };
+  }
+
+  for (const exclusionKey of excludeKeys) {
+    if (exclusionKey.includes(key)) {
+      return { exclude: true, exclusionKey: exclusionKey };
```

Comment:
This only returns the exclusion key as soon as a match is found so even if there is another one for that key it won’t see it